src/Hl7.Fhir.ElementModel/TypedElementOnSourceNode.cs

```diff
@@ -247,6 +251,24 @@ private IEnumerable<TypedElementOnSourceNode> enumerateElements(Dictionary<strin
                     if (_settings.ErrorMode != TypedElementSettings.TypeErrorMode.Passthrough)
                         continue;
                 }
+                var isCcdaXhtmlNode = false;
+
+                //Special condition for ccda.
+                //If we encounter a xhtml node in a ccda document we will flatten all childnodes
+                //and use there content to build up the xml.
+                //The xml will be put in this node and children will be ignored.
+                if (instanceType == xHtmlInstancetype && info.Representation == XmlRepresentation.CdaText)
+                {
+                    string xml = string.Empty;
+                    foreach (var xmlChild in scan.Children())
+                    {
+                        var ix = xmlChild.Annotation<IXHtml>();
+                        xml += ix?.OriginalElement;
+
+                    }
+                    isCcdaXhtmlNode = true;
+                    source = SourceNode.Valued(scan.Name, xml);
```

Comment:
Try to yield here already. To do this, then move the prettypath logic before this block

Reply:
Did this while adding a continue to make this double yield working. Also moved prettypath logic before and removed isCcdaXhtmlNode since its not necessary any more.